src/gamestates/cMainMenuGameState.cpp

```diff
@@ -50,10 +41,17 @@ cMainMenuGameState::cMainMenuGameState(cGame &theGame) : cGameState(theGame), te
 
     int creditsX = (game.m_screenW / 2) - buttonWidth;
     const cRectangle &creditsRect = cRectangle(creditsX, 0, buttonWidth, buttonHeight);
-    gui_btn_credits = new cGuiButton(textDrawer, creditsRect, "CREDITS", buttonKinds);
-    gui_btn_credits->setTextAlignHorizontal(buttonTextAlignment);
-    gui_btn_credits->setOnLeftMouseButtonClickedAction(new cGuiActionToGameState(GAME_CREDITS, true));
 
+    gui_btn_credits = GuiButtonBuilder()
+            .withRect(creditsRect)        
+            .withLabel("CREDITS")
+            .withTextDrawer(&textDrawer)    
+            .withTheme(GuiTheme::Light())
+            .withKind(GuiRenderKind::TRANSPARENT_WITHOUT_BORDER)
+            .onClick([this]() {
```

Comment:
oh this is nice, I've never seen lambda's in C++ before like this. Seems more compact than last time I saw it.